app/controllers/main/routes.py

```diff
@@ -45,42 +46,76 @@ def events(selectedTerm=None):
+                                       "completedTraining" : (username in trainedParticipants(program)),
+                                       "isNotBanned" : isEligibleForProgram(program, username),
+                                       "banNote": noteForDict})
+         return render_template ("/main/volunteerProfile.html",
+            programs = programs,
+            interests = interests,
+            programsInterested = programsInterested,
+            # trainingChecklist = trainingChecklist,
+            upcomingEvents = upcomingEvents,
+            eligibilityTable = eligibilityTable,
+            volunteer = User.get(User.username == username),
+            user = g.current_user)
+    abort(403)
+
+
+@main_bp.route('/banUnbanUser/<program_id>/<username>', methods=['POST'])
+def banUser(program_id, username):
+    """
+    This function updates the ban status of a username either when they are banned or unbanned from a program.
+    program_id: the primary id of the program the student is being banned or unbanned from
+    username: unique value of a user to correctly identify them
+    """
+    postData = request.form
+    banNote = postData["note"] # This contains the note left about the change
+    banOrUnban = postData["banOrUnban"] # Contains "Ban" or "Unban" to determine whether to ban or unban the user
+    banEndDate = postData["endDate"] # Contains the date the ban will no longer be effective
     try:
-        profileUser = User.get(User.username == username)
-        upcomingEvents = getUpcomingEventsForUser(username)
-        programs = Program.select()
-        interests = Interest.select().where(Interest.user == profileUser)
-        interests_ids = [interest.program.id for interest in interests]
-        rsvpedEventsList = EventRsvp.select().where(EventRsvp.user == profileUser)
-        rsvpedEvents = [event.event.id for event in rsvpedEventsList]
-
-        return render_template('/volunteer/volunteerProfile.html',
-                               title="Volunteer Interest",
-                               user = profileUser,
-                               programs = programs,
-                               interests = interests,
-                               interests_ids = interests_ids,
-                               upcomingEvents = upcomingEvents,
-                               rsvpedEvents = rsvpedEvents)
+        return banUnbanUser(banOrUnban, program_id, username, banNote, banEndDate, g.current_user)
     except Exception as e:
         print(e)
-        return "Error retrieving user profile", 500
+        return "Error Updating Ban", 500
+
 
-@main_bp.route('/deleteInterest/<program_id>', methods = ['POST'])
-@main_bp.route('/addInterest/<program_id>', methods = ['POST'])
-def updateInterest(program_id):
+@main_bp.route('/deleteInterest/<program_id>/<username>', methods = ['POST'])
+@main_bp.route('/addInterest/<program_id>/<username>', methods = ['POST'])
+def updateInterest(program_id, username):
```

Comment:
I would reorganize this route to put username first, since that's how our other user-related routes are. like `<username>/delete_interest/<program_id>`

Reply:
The ban/unban routes should follow the same pattern